Revert "Add files via upload"

dispersiyadepressii opened the pull request:
Reverts dispersiyadepressii/funcforbitc#1

dispersiyadepressii committed:
Revert "Add files via upload"

dispersiyadepressii committed:
Add files via upload

All comments have been taken into account, tests have been added in file test_functions

dispersiyadepressii committed:
Add files via upload

create tests for Benefit func, fixed mistakes

kozlovb commented on functions.py:
```diff
@@ -1,34 +1,86 @@
     with open(nameoffile,'r') as f:
-        arrayline = f.readlines()
-        return arrayline
+        arrdata = f.readlines()
```
We could also add here a logic that splits data in : quantity, data and price while casting types to the relevant ones. 

I would suggest to have a structure with these fields.
so something like 
class State
_date_in_sec_epoch
_price_eur
_qty
.....

dispersiyadepressii replied:
Thank you for your comment, will add the class in the next version

kozlovb commented on functions.py:
```diff
@@ -1,34 +1,86 @@
+    for i in range(0, len(arrdata)):
+        arrline  = arrdata[i].split(',', 2)
+        if float(arrline[1]) < previousdate:
+            print("array doesnt ordered by date")
```
small correction - "array IS NOT ordered by date"

dispersiyadepressii replied:
Thank you

kozlovb commented on functions.py:
```diff
@@ -1,34 +1,86 @@
+            print("previousdate:", previousdate, "date:", arrline[1]) 
+            return 1
+        previousdate = float(arrline[1])
+    return 0
```
let's return boolean , so true ot false. It would be slightly more intuitive.

dispersiyadepressii replied:
thanks for your comment, it will be corrected

kozlovb commented on main.py:
```diff
@@ -1,15 +1,29 @@
+textlist = functions.readStateFile(filename)
+
+if functions.checkOrderByDate(textlist) == 1:
+    textlist = functions.SortByDate(textlist)
```
it's a good idea to have a possibility to sort data. However in my particular case I'd like this programm to fail. As if I have an incorrect state means I have to investigate on how it happened. So, lets keep sorting function but for now pls deactivate it.

dispersiyadepressii replied:
okay, will add some choice for situation with unsorted data, thank you

kozlovb commented on main.py:
```diff
@@ -1,15 +1,29 @@
+print(functions.Benefit(sum, price, textlist))
 
-functions.AddLine()
+#functions.AddLine()
```
pls remove

kozlovb commented on functions.py:
```diff
@@ -1,34 +1,86 @@
 #how much is not taxed
-def SellWithoutTax():
-    arrayline = ReadLines('new_register.txt')
+def SellWithoutTax(arrdata):
```
I'd like to be able to provide arrdata in a human readable format. 
Could you pls specify the format that I would need to provide ass well as a time zone.

kozlovb commented on functions.py:
```diff
@@ -1,34 +1,86 @@
-        arr = arrayline[i].split(',', 2)
+    for i in range(0, len(arrdata)):
+        arrline  = arrdata[i].split(',', 2)
+        if (time.time() - float(arrline[1])) >= SEC_IN_YEAR:
```
Remember there are 2 conditions for BTC to be sold without tax.
One is held more than a year and the other is that no profit is made.
I.e. IF price is now 16K But I bought it at 30K I can definitely sell int without taxes.

But be careful if say now price is 16K, but I bought 1BTC for 30K 6 month ago and then 1BTC for 8K 2 months ago I still don't want to count the ones I bought or 30K. Cause I cannot sell them before I sell the ones for 8K.

kozlovb commented on functions.py:
```diff
@@ -1,34 +1,86 @@
-    print("benefit:", sum*price - totalprice)
-    return (sum*price - totalprice)
+#counting benefit 
+def Benefit(sum, price, arrdata):
```
May be a bit more details on what this function does ? 

Basically, I might be interested in something like 
[benefit, taxable benefit] output.

kozlovb commented on functions.py:
```diff
@@ -40,10 +92,10 @@ def AddLine():
-    print(type(text))
-    f.write(text)
-    f.close()
+    with open('register_copi.txt','a') as f:
```
copy

kozlovb commented on functions.py:
```diff
@@ -1,34 +1,86 @@
-    return (sum*price - totalprice)
+#counting benefit 
+def Benefit(sum, price, arrdata):
+    cantsell = sum
```
please check functions name Benefit -> benefit and in other places if any

kozlovb commented on test_functions.py:
```diff
@@ -0,0 +1,83 @@
+    result = functions.readStateFile(test_file)
+
+    # Assert that the result is equal to the expected output
+    assert result == ["Test data"]
```
Let's do something more realistic for this test.
Like a line from a actual state test file.

kozlovb commented on test_functions.py:
```diff
@@ -0,0 +1,83 @@
+
+def test_SellWithoutTax():
+    #Test case: list is empty
+    arrdata = []
```
nice

kozlovb commented on test_functions.py:
```diff
@@ -0,0 +1,83 @@
+    arrdata = ["10.0, 1598346000, somedata"]
+    assert functions.SellWithoutTax(arrdata) == 10.0
+
+    #Test case: input array with one element, where the time is less than SEC_IN_YEAR
```
Actually, the "time flies". Be careful time goes on and in one year this test will start to fail. Please correct

kozlovb commented on test_functions.py:
```diff
@@ -0,0 +1,83 @@
+    assert functions.SellWithoutTax(arrdata) == 0.0
+
+    #Test case:an input array with multiple elements, where some elements can be sold without tax
+    arrdata = ["5.0, 1798346000, data1", "20.0, 1638346000, data2", "15.0, 1645862400, data3"]
```
price is one of the factors that determines if a token can be sold without tax. Pls introduce prices and checks that reflect this

kozlovb reviewed (commented):
good ! Keep on.

dispersiyadepressii committed:
Add files via upload

tests for the functions are not ready yet, a function for sale has been added, AddNewLine has been renamed PurchaseNewLine, after the sale of coins, a record of the transaction will be made in a separate file ( it will be created if this does not exist yet )

dispersiyadepressii committed:
Add files via upload

dispersiyadepressii committed:
Add files via upload

kozlovb commented on functions.py:
```diff
@@ -1,49 +1,141 @@
+        f.close()
+    print("ok")
+
+def SellDelLines(arrdata, price, nameoffile):
```
arrdata - state 
and docs pls - what are the params , what this function outputs etc
